kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala

```diff
@@ -198,6 +224,38 @@ class BatchJobSubmission(
     }
   }
 
+  private def monitorSubmittedApp(appId: String): Unit = {
+    info(s"Monitoring submitted $batchType batch application: $appId")
+    if (applicationStatus.isEmpty) {
+      applicationStatus = currentApplicationState
+    }
+    if (applicationStatus.isEmpty) {
+      info("The batch application not found, assume that it has finished.")
+    } else if (applicationFailed(applicationStatus)) {
+      throw new RuntimeException("Batch job failed:" + applicationStatus.get.mkString(","))
+    } else {
+      val waitSubmittedAppCompletion = builder match {
+        case sbp: SparkBatchProcessBuilder => sbp.waitAppCompletion()
+        case _ => false
+      }
+
+      if (!waitSubmittedAppCompletion) {
+        info(s"The batch application has been submitted successfully:" +
```

Comment:
it only affect the release time of client, but the engine may still active ? I think we should also check the application status.

Reply:
yes, we have checked `applicationFailed` in other branch.
So now it is in `RUNNING` or TERMINATED state.